primitives/arithmetic/src/per_things.rs

```diff
@@ -67,7 +67,8 @@ pub trait PerThing:
 		let b = Self::ACCURACY;
 		// if Self::ACCURACY % 100 > 0 then we need the correction for accuracy
 		let c = rational_mul_correction::<Self::Inner, Self>(b, a, 100.into(), Rounding::Nearest);
-		Self::from_parts(a / 100.into() * b + c)
+		let base: Self::Inner = 100.into();
+		Self::from_parts(a / base.saturating_mul(b).saturating_add(c))
```

Comment:
but this is changing the operation order, then the denominator overflows:
`<PerU16 as PerThing>::from_percent(100) == 0`

```suggestion
		let base: Self::Inner = 100.into();
		Self::from_parts((a / base).saturating_mul(b).saturating_add(c))
```
Maybe you could also improve test to assert on the resulting value

Reply:
damn I didn't even try that 🤦  thanks